### PR TITLE
Add doctests to interpolation_search.py

### PR DESCRIPTION
**Pull request by Ihor-Pryyma**
### Describe your change:
Contributes to https://github.com/TheAlgorithms/Python/issues/9943

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Review by Abonimohan — APPROVED** [review]

**Review by Abonimohan — CHANGES_REQUESTED** [review]
Line 11-12 please check it out

**Review by imSanko — CHANGES_REQUESTED** [review]
`value not in ("#", "")`. Use a `set` if the elements are hashable.

**Comment by Ihor-Pryyma** [comment]
> `value not in ("#", "")`. Use a `set` if the elements are hashable.

but that file is not in my PR

File: searches/interpolation_search.py [hunk 1]
```diff
@@ -3,13 +3,34 @@
...
+
+    Returns:
+        int: The index of the found item, or None if the item is not found.
+    Examples:
+    >>> interpolation_search([1, 2, 3, 4, 5], 2)
+    1
+    >>> interpolation_search([1, 2, 3, 4, 5], 4)
+    3
+    >>> interpolation_search([1, 2, 3, 4, 5], 6)
+    >>> interpolation_search([], 1)
```
Comment (cclauss) [inline searches/interpolation_search.py:22]:
```suggestion
    >>> interpolation_search([1, 2, 3, 4, 5], 6) is None
    True
    >>> interpolation_search([], 1) is None
    True
```
Please repeat below.

**Comment by cclauss** [comment]
`web_programming/emails_from_url.py` was fixed in a separate pull request.

File: searches/interpolation_search.py (hunk 1 above)
Comment (Ihor-Pryyma) [reply searches/interpolation_search.py]: fixed

**Review by imSanko — COMMENTED** [review]
Looks good to be @cclauss will let know better to move forward with this.

**Review by cclauss — APPROVED** [review]
Thanks!

**Pull request merged**